dev/io.openliberty.cdi.4.0.internal.weld/src/io/openliberty/cdi40/internal/weld/LiteExtensionArchive.java

```diff
@@ -0,0 +1,181 @@
+/*******************************************************************************
+ * Copyright (c) 2022 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package io.openliberty.cdi40.internal.weld;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Supplier;
+
+import org.jboss.weld.lite.extension.translator.LiteExtensionTranslator;
+
+import com.ibm.websphere.csi.J2EEName;
+import com.ibm.websphere.ras.Tr;
+import com.ibm.websphere.ras.TraceComponent;
+import com.ibm.ws.cdi.CDIException;
+import com.ibm.ws.cdi.internal.archive.AbstractCDIArchive;
+import com.ibm.ws.cdi.internal.interfaces.Application;
+import com.ibm.ws.cdi.internal.interfaces.ArchiveType;
+import com.ibm.ws.cdi.internal.interfaces.CDIArchive;
+import com.ibm.ws.cdi.internal.interfaces.CDIRuntime;
+import com.ibm.ws.cdi.internal.interfaces.ExtensionArchive;
+import com.ibm.ws.cdi.internal.interfaces.Resource;
+import com.ibm.ws.cdi.internal.interfaces.ResourceInjectionBag;
+import com.ibm.ws.runtime.metadata.MetaData;
+
+import jakarta.enterprise.inject.build.compatible.spi.BuildCompatibleExtension;
+
+/**
+ * Extension archive which adds the Weld LiteExtensionTranslator
+ */
+public class LiteExtensionArchive extends AbstractCDIArchive implements ExtensionArchive {
+    private static final TraceComponent tc = Tr.register(LiteExtensionArchive.class);
+    private final ClassLoader bceClassLoader;
+    private final List<Class<? extends BuildCompatibleExtension>> buildCompatibleExtensions;
+
+    public LiteExtensionArchive(CDIRuntime cdiRuntime, ClassLoader bceClassLoader, List<Class<? extends BuildCompatibleExtension>> buildCompatibleExtensions) {
+        super(LiteExtensionTranslator.class.getName(), cdiRuntime);
+        this.bceClassLoader = bceClassLoader;
+        this.buildCompatibleExtensions = buildCompatibleExtensions;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public J2EEName getJ2EEName() throws CDIException {
+        return null;
```

Comment:
Appears that returning null is the correct thing to do but maybe we need a comment to explain why?

Reply:
Probably a good idea, I think I just copied from the `ProbeExtensionArchive`.

Reply:
For this particular method, the javadoc on `CDIArchive` says it can return `null` if there's no J2EE name, but for many of the others it doesn't, so I'm not sure if `null` is a valid return value or if the methods are just not called on these archives for some reason.

Does warrant some further investigation, I'll see if I can look at the existing code and update the `CDIArchive` javadoc.

Reply:
The handling of a null J2EEName doesn't seem 100% safe/consistent. It is a shame that we need it to be a J2EEName and not a more free form string ID.

Reply:
From what I can see, it's either null-checked or called on module archives.

Reply:
OK, this PR does not change that situation. So _if_ it is a problem we can deal with it separately.

Reply:
I've updated the Javadoc on CDIArchive based on what the existing code already does to indicate under what circumstances methods are allows to return `null`.